python/python_fastapi_server/routers/edr.py

```diff
@@ -0,0 +1,920 @@
+                        edr_collection.attrib.get("z_interval"),
+                        "parameters": [
+                            inst.strip()
+                            for inst in edr_collection.text.strip().split(",")
+                        ],
+                    }
+        except ParseError:
+            pass
+    return edr_collections
+
+
+def get_edr_collections_for_datasetOLD(
+        dataset: str, dataset_dir: str = os.environ["ADAGUC_DATASET_DIR"]):
+    logger.info(">>get_edr_collections_for_dataset %s", dataset)
+    filename = f"{dataset_dir}/{dataset}.xml"
+
+    edr_collections = {}
+    tree = parse(filename)
+    root = tree.getroot()
+    for ogcapi_edr in root.iter("OgcApiEdr"):
+        for edr_collection in ogcapi_edr.iter("EdrCollection"):
+            edr_collections[edr_collection.attrib.get("name")] = {
+                "dataset":
+                dataset,
+                "name":
+                edr_collection.attrib.get("name"),
+                "url":
+                "http://localhost:8000/wms",
+                "time_interval":
+                edr_collection.attrib.get("time_interval"),
+                "z_interval":
+                edr_collection.attrib.get("z_interval"),
+                "parameters": [
+                    inst.strip()
+                    for inst in edr_collection.text.strip().split(",")
+                ],
+            }
+    return edr_collections
+
+
+def get_point_value(
+    dataset: str,
+    instance: str,
+    coords: list[float],
+    parameters: str,
+    t: str,
+    z: str = None,
+    extra_dims="",
+):
+    urlrequest = (
+        f"SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPointValue&CRS=EPSG:4326"
+        f"&DATASET={dataset}&QUERY_LAYERS={parameters}"
+        f"&X={coords[0]}&Y={coords[1]}&INFO_FORMAT=application/json")
+    if t:
+        urlrequest += f"&TIME={t}"
+
+    if instance:
+        urlrequest += f"&DIM_reference_time={instance_to_iso(instance)}"
+    if z:
+        urlrequest += f"&ELEVATION={z}"
+    if extra_dims:
+        urlrequest += extra_dims
+
+    logger.info("URL: %s", urlrequest)
+    status, response = call_adaguc(url=urlrequest.encode("UTF-8"))
+    logger.info("status: %d", status)
+    if status == 0:
+        return response.getvalue()
+    return None
+
+
+@edrApiApp.get(
+    "/collections/{collection_name}/position",
+    response_model=Coverage,
+    response_model_exclude_none=True,
+)
+async def get_collection_position(
+        collection_name: str,
+        request: Request,
+        coords: str,
+        datetime: Optional[str] = None,
+        parameter_name: str = Query(alias="parameter-name"),
+):
+    allowed_params = ["coords", "datatime", "parameter-name"]
+    extra_params = [k for k in request.query_params if k not in allowed_params]
+    extra_dims = ""
+    if len(extra_params):
+        for extra_param in extra_params:
+            extra_dims += f"&DIM_{extra_param}={request.query_params[extra_param]}"
```

Comment:
What are the extra parameters used for exactly? Do you have an example?

Reply:
They are custom dimensions or the data. Because they are custom, they can't be coded into the method's parameter list by name, they have to be extracted from the query parameter.

I renamed them to custom_dims